quarkus/service/src/test/java/org/apache/polaris/service/quarkus/catalog/BasePolarisCatalogTest.java

```diff
@@ -1459,6 +1462,46 @@ public void testDropTableWithPurgeDisabled() {
         .hasMessageContaining(PolarisConfiguration.DROP_WITH_PURGE_ENABLED.key);
   }
 
+  @Test
+  public void testRefreshIOWithCredentialsAndInternalProperties() {
```

Comment:
Not sure how this actually _refreshes_ credentials.

Anyway, having this test is good. However, I'd prefer to also have a dedicated test class for `DefaultFileIOFactory` and do extensive testing there, also with multiple realms?